source/common/network/io_socket_handle_impl.cc

```diff
@@ -168,7 +190,11 @@ Api::IoCallUint64Result IoSocketHandleImpl::sendmsg(const Buffer::RawSlice* slic
     message.msg_control = nullptr;
     message.msg_controllen = 0;
     const Api::SysCallSizeResult result = os_syscalls.sendmsg(fd_, &message, flags);
-    return sysCallResultToIoCallResult(result);
+    auto io_result = sysCallResultToIoCallResult(result);
+    if (io_result.wouldBlock() && file_event_) {
+      file_event_->registerEventIfEmulatedEdge(Event::FileReadyType::Write);
+    }
+    return io_result;
```

Comment:
they are used for quic/udp listeners. I added them because they caused various tests to hang

Reply:
Seems odd that UDP was operating in edge mode...  Hmm, I guess it is necessary to avoid getting overwhelmed by write events in Level mode.  Ok, make sense.